src/settings-ui/Settings.UI/ViewModels/PluginMetadataViewModel.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.PowerToys.Settings.UI.ViewModels
+{
+    public class PluginMetadataViewModel : INotifyPropertyChanged
```

Comment:
This and the multiple DataTemplateSelectors feel like a lot of overkill.

Wouldn't a simple StackPanel and setting the visibility on the website hyperlink/text be sufficient?

Reply:
Was thinking the same while reviewing it

Reply:
The problem is that with only setting visibility the page crashes on invalid uri format. We need a way to no create the ui control if the uri is wrong.

Alternatively we can init the hyperlink with a dummy url (localhost) and hidden if the uri is wrong.

What do you think?

Reply:
Would having logic in the viewmodel to check the URL and either return null or string.empty (if not in the right format work?

And then show/hide the textblock/hyperlink based on if it's null/empty or an actual URL?

Coming back to your earlier mentioning about the Toolkits MetadataControl, what are the limitations there?

Reply:
I don't think that we can try init a hyperlink control with null. And I definitely tested that it crashed on initializing with empty string. And it is always initialize as long as it technically exists. So either we use template selection or we use a dummy url (e.g. http://localhost).

Regarding the metadata control for me it seem that we can change the separator. But we can't format the text layout for hyperlinks and texts. You only can set the values for the metadata items. And I personally like to keep the current layout with tooltip, bold text and accent color.

Reply:
Yeah, it feels like setting it to localhost or microsoft.com might be a better solution to not introduce a ton of code for a minor feature. Also suggested here:

https://stackoverflow.com/questions/70370268/crash-when-using-hyperlinkbutton-with-empty-navigateuri

I'm personally not against changing the fontweights if we get better maintainable code in return, but I'll leave that up to you :)!